src/test-integration/java/com/bettercloud/vault/api/LogicalTests.java

```diff
@@ -297,12 +293,10 @@ public void testDeleteExceptionMessageIncludesErrorsReturnedByVault() throws Vau
      * @throws VaultException
      */
     @Test
-    public void testListExceptionMessageIncludesErrorsReturnedByVault() throws VaultException {
-        expectedEx.expect(VaultException.class);
-        expectedEx.expectMessage("permission denied");
-
+    public void testListPermissionDeniedReturnedByVault() throws VaultException {
         final Vault vault = container.getVault(NONROOT_TOKEN);
-        vault.logical().list("secret/null");
+        List<String> list = vault.logical().list("secret/null");
+        assertEquals(list.size(), 0);
```

Comment:
@bklawans not sure this is desirable, see https://github.com/BetterCloud/vault-java-driver/pull/176#issuecomment-524583812

Reply:
Since @steve-perkins bumped the major version and I have fixed up CI. Let's go crazy and break the API 😆

Reply:
`getListData` seems responsible to me as you already have `getData` but that is not the same as `read` so perhaps we need a new response class.